plugins/common/opcua/client.go

```diff
@@ -63,30 +61,53 @@ func (o *OpcUAClientConfig) Validate() error {
 }
 
 func (o *OpcUAClientConfig) validateOptionalFields() error {
-	validFields := []string{"DataType"}
-	return choice.CheckSlice(o.OptionalFields, validFields)
+	for i, field := range o.OptionalFields {
+		switch field {
+		case "DataType":
+			// Valid optional field
+		default:
+			return fmt.Errorf("invalid optional field %q at index %d, expected one of: [DataType]", field, i)
+		}
+	}
+	return nil
 }
 
 func (o *OpcUAClientConfig) validateEndpoint() error {
 	if o.Endpoint == "" {
-		return errors.New("endpoint url is empty")
+		return &EndpointError{
+			Endpoint: o.Endpoint,
+			Err:      fmt.Errorf("%w: endpoint URL is empty", ErrInvalidEndpoint),
+		}
 	}
 
 	_, err := url.Parse(o.Endpoint)
 	if err != nil {
-		return errors.New("endpoint url is invalid")
+		return &EndpointError{
+			Endpoint: o.Endpoint,
+			Err:      fmt.Errorf("%w: %w", ErrInvalidEndpoint, err),
+		}
 	}
 
 	switch o.SecurityPolicy {
-	case "None", "Basic128Rsa15", "Basic256", "Basic256Sha256", "auto":
+	case "None", "Basic128Rsa15", "Basic256", "Basic256Sha256", "Aes128_Sha256_RsaOaep", "Aes256_Sha256_RsaPss", "auto":
+		// Valid security policy
 	default:
-		return fmt.Errorf("invalid security type %q in %q", o.SecurityPolicy, o.Endpoint)
+		validPolicies := []string{"None", "Basic128Rsa15", "Basic256", "Basic256Sha256", "Aes128_Sha256_RsaOaep", "Aes256_Sha256_RsaPss", "auto"}
```

Comment:
Why do you change this and especially why are you adding new policies? This should not be done in a cleanup PR!

Reply:
You're right @srebhan I'll remove the security policy changes from this cleanup PR and handle/discuss new policies separately. Thanks for catching that!